mne/preprocessing/ica.py

```diff
@@ -1367,12 +1379,12 @@ def _check_exclude(self, exclude):
         if exclude is None:
             return list(set(self.exclude))
         else:
-            return list(set(self.exclude + exclude))
+            # Allow both self.exclude and exclude to be array-like:
+            return list(set(self.exclude).union(set(exclude)))
 
     def _apply_raw(self, raw, include, exclude, n_pca_components, start, stop):
         """Aux method."""
         _check_preload(raw, "ica.apply")
-        exclude = self._check_exclude(exclude)
```

Comment:
why removing all these calls to _check_exclude?

Reply:
... because this function is called in _pick_sources, which is called by all the apply_... funcs. Since _pick_sources does the exclude-relevant work I found it more consistent and future-proof to call _check_exclude there.